### DISCUSSION TITLE
Discord server permissions and new channels setup proposal

### DISCUSSION THREAD

**Discussion by tamagit80** [SecretDecks/Documentation#123] [Ideas]
### Why
After analizing the actual state of our Discord setup I noticed that is not following some usefull standards.
<hr>

**_Actual Channels setup_**
- Missing a stage channel
- Missing announcement channel with follow button
- Missing a public voice chat channel

_**Actual persmissions setup:**_
- Categories don't have a permissions setup, only the single channels.
- Single channels are not synced to the parent category.
- There is no hierarchical structure.
<hr>

### _**New Channels Proposal:**_ 
The symbol "+" indicates a new channel

**Information**
- Welcome
- Introduction
- Rank Up
- Voice chat +

**News** + (public)
- Announcements -> github announcments channel with the follow button +
- Events +

**DisCo Channels/Shell Channel**
- Voting
- Waterhole
- Market-things
- Artist Coordination
- Creative +
- Support +
- Off topic +
- Voice chat +

**Shell Channels**
Incorporated in DisCo

**Core Channels**
- Organization Core
- Dev Core
- Creative Core
- Support Core
- Legal Core
- Voice chat +

**Voice chat**
Stage Channel +
<hr>

### **_New Permissions Proposal_**
<hr>

**INFORMATION CATEGORY PERMISSIONS**
Everyone role can READ in welcome and introduction
Everyone role can WRITE in introduction
A bot can answer and ask to react, so ppl don't have to go back to the welcome channel
After the reaction they will get Frontgate permissions.

Note:
Maybe we could toggle permissions to see message history in the introduction page and add a small intro?
It could make it easier to understand and the page cleaner.

Rank-up channel shows up.
Frontgate can READ and REACT. Everyone has no permissions anymore!
After the reaction they will get Shell permissions.

We can add a voice chat with special permissions.
Ex: Everyone role should get permission from higher roles to activate the mic.
<hr>

**DISCO SHELL CHANNELS PERMISSIONS**
Frontgate and Everyone no permissions

**General Category Permissions:**
View channel: YES
Manage channel: NO
Manage Permissions: NO
Manage WebHook: NO

**Membership Permissions:**
Create Invite to the specific Channel: YES

**Text channel permissions:**
Send Messages: YES
Send messages in thread: YES
Create Public threads: YES
Create Private threads: YES
Embed Links: YES
Attach files: YES
Add reactions: NO
Use external Emoji: NO
Use external Stickers: NO
Mention @everyone, @here and all role: YES
Menage Messages: NO
Menage Threads: NO
Read messages history: YES
Send txt-to-speech messages: YES
Use application command: YES

**Voice channel permissions:**
Connect: YES
Speak: YES
Video: YES
Use voice activity: YES
Priority speaker: NO
Mute Members: NO

**Stage channel permissions:**
Request to speak: YES
<hr>

**CORE CHANNELS PERMISSIONS**
Shell, Frontgate and Everyone no permissions

**General Category Permissions:**
View channel: YES
Manage channel: NO
Manage Permissions: NO
Manage WebHook: NO

**Membership Permissions:**
Create Invite to the specific Channel: YES

**Text channel permissions:**
Send Messages: YES
Send messages in thread: YES
Create Public threads: YES
Create Private threads: YES
Embed Links: YES
Attach files: YES
Add reactions: NO
Use external Emoji: NO
Use external Stickers: NO
Mention @everyone, @here and all role: YES
Menage Messages: NO
Menage Threads: NO
Read messages history: YES
Send txt-to-speech messages: YES
Use application command: YES

**Voice channel permissions:**
Connect: YES
Speak: YES
Video: YES
Use voice activity: YES
Priority speaker: NO
Mute Members: NO

**Stage channel permissions:**
Request to speak: YES
<hr>

**_ADMIN_** role has all the permissions
Everything shell/core can't do admin can but only after a community vote.
<hr>

### **_Question_**
Should we consider using hierarchical permissions? ex: Core has all the permissions of the shell channel, also the ones shell role doesn't have (exluding developing permissions ex: webhook, add reactions ecc.)

**Comment by DanM3rcurius** [comment]
First question: Why should we follow standards?
If the standard makes sense for our case we can use it but not without giving a rationale for it.

> Missing a stage channel: 

for what

> Missing announcement channel with follow button
not yet needed as we dont have public announcements.

> Missing a public voice chat channel
not needed. What's the value added or use case? What's the target audience?

This is a great collection and good start for the Discord Adaptation epic. I'll link this issue when writing the Epic and Stories.

We need to remodel Discord anyway to accommodate the visitors from the website.

There will be separate areas only accessible for artists or tarot heads (one channel each); our community manager will welcome them and show them channels relevant for them individually, If channels require Core or shell role, the community manager will explain or show them how to achieve those roles.
I also agree with adding new channels, however not sure if we should create channels without explicit need for them.
Also since now many channels are not used, so removing them and then adding channels later when needed (for example if a long thread is emerging from a channel and goes on for longer than 5 days, this will become a channel). 
Additionally a channel should be retired if not used for over a month.
A kind of organic channel structure. Since discussions and ideation and developing of complex systems is not sth that can be moulded or standardised well without killing some creativity. 
That's why Github should be more standardised and Discord a bit more adaptable.

Thoughts? @tamagit80 @stephen-rowan @wlstylr 

Thoughts?

**Comment by DanM3rcurius** [comment]
> A bot can answer and ask to react, so ppl don't have to go back to the welcome channel
After the reaction they will get Frontgate permissions.

This is not possible afaik. if it's possible and doable in 1 hour. please do so. else we stick to the process now as its one of the few things that work ^^

**Comment by DanM3rcurius** [comment]
> Maybe we could toggle permissions to see message history in the introduction page and add a small intro?

security risk: bots can crawl the content then, which is a bit too sensitive i feel.

**Comment by tamagit80** [comment]
Please find my comment below using
```diff
@@ purple and bold @@
```

> First question: Why should we follow standards? If the standard makes sense for our case we can use it but not without giving a rationale for it.
```diff
@@ With the term standard i mean usability (user experience), compatibility and functionalities @@
```
> 
> > Missing a stage channel:
> 
> for what.
```diff
@@ I think it will be easier for people to understand during the session and to manage for mediators especially when there will be more people joining. @@
```
> 
> > Missing announcement channel with follow button
> > not yet needed as we dont have public announcements.
> 
> > Missing a public voice chat channel
> > not needed. What's the value added or use case? What's the target audience?
```diff
@@ Can be a place to help people who doesn't know anything about discord and technology. Direct and voice chat are now very important for the first support. @@
```
> 
> This is a great collection and good start for the Discord Adaptation epic. I'll link this issue when writing the Epic and Stories.
> 
> We need to remodel Discord anyway to accommodate the visitors from the website.
> 
> There will be separate areas only accessible for artists or tarot heads (one channel each); our community manager will welcome them and show them channels relevant for them individually, If channels require Core or shell role, the community manager will explain or show them how to achieve those roles. 
```diff
@@ Voice chat could be helpful for that too @@
```

I also agree with adding new channels, however not sure if we should create channels without explicit need for them. Also since now many channels are not used, so removing them and then adding channels later when needed (for example if a long thread is emerging from a channel and goes on for longer than 5 days, this will become a channel). 
```diff
@@ In the proposal i removed completely shell channels @@
```

Additionally a channel should be retired if not used for over a month. A kind of organic channel structure. Since discussions and ideation and developing of complex systems is not sth that can be moulded or standardised well without killing some creativity.
```diff
@@ Interesting. we should find a way to create an archive of retired channels. @@
```

That's why Github should be more standardised and Discord a bit more adaptable.
> 
> Thoughts? @tamagit80 @stephen-rowan @wlstylr
> 
> Thoughts?

**Comment by tamagit80** [comment]
> > A bot can answer and ask to react, so ppl don't have to go back to the welcome channel
> > After the reaction they will get Frontgate permissions.
> 
> This is not possible afaik. if it's possible and doable in 1 hour. please do so. else we stick to the process now as its one of the few things that work ^^
```diff
@@ We would just need an auto-reply, i think it can be done also without a bot. @@
```

**Comment by tamagit80** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/SecretDecks-Website-Relaunch--Founders-Drop-615413e56a2e7e0012d7f67b/issues/SecretDecks/Documentation/121?planning-poker) with ZenHub @DanM3rcurius @evolayak @sarafanciulli @stephen-rowan @wlstylr

**Comment by DanM3rcurius** [comment]
this is not a a story. Only stories can be estimated. 
It's not clear what is done and why, also how to know when this is done?

**Comment by DanM3rcurius** [comment]
## Format of a Story

**Story: refers to "user story" **

what is the **person** interacting (with the feature or feature set) **wanting**, **expecting**, **seeing** and **experiencing**
or
**How** is the **person doing what and why** from the I-perspective

**Comment by stephen-rowan** [comment]
The person is wanting a bit of fun ?

**Comment by tamagit80** [comment]
> this is not a a story. Only stories can be estimated. It's not clear what is done and why, also how to know when this is done?

```diff
@@ I thought every issue can be estimated as per option of zenhub. @@

@@ I can modify the issue and explain better what, why and when it will be done. It's anyway just a way to discuss about it. @@
```

**Comment by wlstylr** [comment]
> First question: Why should we follow standards? If the standard makes sense for our case we can use it but not without giving a rationale for it.
> 
> > Missing a stage channel:
> 
> for what
> 
> > Missing announcement channel with follow button
> > not yet needed as we dont have public announcements.
> 
> > Missing a public voice chat channel
> > not needed. What's the value added or use case? What's the target audience?
> 
> This is a great collection and good start for the Discord Adaptation epic. I'll link this issue when writing the Epic and Stories.
> 
> We need to remodel Discord anyway to accommodate the visitors from the website.
> 
> There will be separate areas only accessible for artists or tarot heads (one channel each); our community manager will welcome them and show them channels relevant for them individually, If channels require Core or shell role, the community manager will explain or show them how to achieve those roles. I also agree with adding new channels, however not sure if we should create channels without explicit need for them. Also since now many channels are not used, so removing them and then adding channels later when needed (for example if a long thread is emerging from a channel and goes on for longer than 5 days, this will become a channel). Additionally a channel should be retired if not used for over a month. A kind of organic channel structure. Since discussions and ideation and developing of complex systems is not sth that can be moulded or standardised well without killing some creativity. That's why Github should be more standardised and Discord a bit more adaptable.
> 
> Thoughts? @tamagit80 @stephen-rowan @wlstylr
> 
> Thoughts?

I'm in favor for an announcements channel.  It will be essential for communication of Founders drop announcements and updates.

Overall, there is a lot of to take in.  I think I'm in agreement that we need to rework the discord channel.  In my mind Discord will become the place where people come to learn about Secret Decks NFT drops and build relationships and community around the Decks and cards they will eventually own. 

Would it make sense to map out what a Discord server channel update should look like in Miro?

**Reply by tamagit80** [reply]
I think it would make sense

**Comment by wlstylr** [comment]
As a visitor to Secret Decks discord, I need to quickly understand where I can participate in the SD ecosystem so that I can be on time for NFT drops, know where the work is happening, and be present to vote for on issues that interest me.  

I think @tamagit80 did an awesome job of outlining improvements to the Discord channel.  My take of a story is above.  @DanM3rcurius

**Reply by DanM3rcurius** [reply]
yes he did! Just reading this now -.-
